File: MiniActivitat3/MiniActivity_3_OP/app/src/main/java/com/example/miniactivity_3_op/MainActivity.java
package com.example.miniactivity_3_op;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        try {
            TextView tv = findViewById(R.id.textView1);
            Bundle data = this.getIntent().getExtras();
            String text = data.getString(Intent.EXTRA_TEXT);
            tv.setText(text);
        } catch(Exception e){}

    }
}

File: MiniActivitat2/app/src/main/java/com/example/miniactivitat2/MainActivity.kt
package com.example.miniactivitat2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView



class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button: Button = findViewById(R.id.button)
        val editText: EditText = findViewById(R.id.editText)
        val editText2: EditText = findViewById(R.id.editText2)

        button.setOnClickListener{
            var intent = Intent(this, Activity2::class.java)
            intent.putExtra("text", editText.text.toString())
            intent.putExtra("rep", editText2.text.toString().toInt())
            startActivityForResult(intent, 1)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (requestCode == 1) {
            var textview: TextView = findViewById(R.id.textView4)
            val returnString = data!!.getStringExtra("MESSAGE")
            textview.text = returnString
        }
    }




}